File: DSA/Sauhard_Semalty_590013630/Day4Question1.py
import sys


def find_minimum(lst: list[int]) -> int | None:
    "Finds the minimum value element from the list."

    # Could use the `min` function here

    if not lst:
        return None

    minimum = lst[0]
    for i in lst:
        if i < minimum:
            minimum = i

    return minimum


def main() -> None:
    "The entry point of the program."

    try:
        n = int(input("Enter the size of the array: "))
    except ValueError:
        print("Invalid size!")
        sys.exit(-1)

    # Check if length is valid
    if n <= 0:
        print("Size must be greater than 0.")
        sys.exit(-1)

    nums: list[int] = []

    print("+--Enter the values--+")
    for _ in range(n):
        try:
            num = int(input("> "))
        except ValueError:
            print("Invalid value!")
            sys.exit(-1)

        nums.append(num)

    # For 'k'
    try:
        k = int(input("Enter the number of elements to shift: "))
    except ValueError:
        print("Invalid number!")
        sys.exit(-1)

    # Check if k is valid
    if not 0 <= k <= n:
        print(f"Number must be between 1 and {n} (inclusive).")
        sys.exit(-1)

    minimum = find_minimum(nums)

    print(minimum)


if __name__ == "__main__":
    main()

File: DSA/DivyanshKumarSingh_590012175/Day_4_Question_1.c
#include <stdio.h>

int Minimum(int nums[], int n) 
{
    int min = nums[0];

    for (int i = 1; i < n; i++) 
    {
        if (nums[i] < min) 
        {
            min = nums[i];
        }
    }

    return min;
}

int main() 
{
    int n;

    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);

    if (n <= 0) 
    {
        printf("Array size must be greater than 0.\n");
        return 1;
    }

    int nums[n];

    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) 
    {
        scanf("%d", &nums[i]);
    }

    int min = Minimum(nums, n);

    printf("The minimum element in the array is: %d\n", min);

    return 0;
}
